optimize.c

```diff
@@ -913,11 +913,11 @@ opt_peep(opt_state_t *opt_state, struct block *b)
 			break;
 
 		case BPF_JGT:
-			v = (unsigned)v > b->s.k;
+			v = v > b->s.k;
 			break;
 
 		case BPF_JGE:
-			v = (unsigned)v >= b->s.k;
+			v = v >= b->s.k;
```

Comment:
OK. I removed that modification at: https://github.com/the-tcpdump-group/libpcap/pull/512/commits/c2d601298b8f57a18fe21ae735360d92e5ebc487